### PR TITLE
connman/wireguard updates

### PR DESCRIPTION
**Pull request by chewitt**
This bumps connman to v1.38 and the WireGuard tools/module packages to their respective latest release tags (the packages are no longer in version sync with each other).

NB: The WireGuard module needs a patch to workaround Linux 5.6 checks the developers added to drive people to the in-kernel module. Once we've bumped projects in master to Linux 5.6 we can enable the in-kernel module everywhere and drop the compat package. In case we forget to do it we'll be reminded with build failures once we reach Linux 5.7, which is why I chose to bump the value instead of simply patching it out.

### CODE REVIEW

File: packages/network/wireguard-linux-compat/patches/wireguard-linux-compat-01-skip-compat_h-version-checks.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/src/compat/compat.h b/src/compat/compat.h
+index 1aa0e7b..751ace7 100644
+--- a/src/compat/compat.h
++++ b/src/compat/compat.h
+@@ -43,7 +43,7 @@
+ #error "WireGuard requires Linux >= 3.10"
+ #endif
+ 
+-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
++#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 7, 0)
```
Comment (MilhouseVH) [inline packages/network/wireguard-linux-compat/patches/wireguard-linux-compat-01-skip-compat_h-version-checks.patch]: The version used in the line above conflicts with the version mentioned in the comment below - which is correct - is it `>= 5, 7, 0` or `>= 5.6`?

Currently `wireguard-linux-compat` doesn't build with 5.6-rc2 (it did build with 5.6-rc1) so I'm assuming the line above should `#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)` and the comment is correct (in which case this patch shouldn't have been modified).

Comment (MilhouseVH) [reply packages/network/wireguard-linux-compat/patches/wireguard-linux-compat-01-skip-compat_h-version-checks.patch]: If we don't want to build this package with kernel >= 5.6 then we shouldn't be building it... causing the build to fail is not the right fix.

Comment (MilhouseVH) [reply packages/network/wireguard-linux-compat/patches/wireguard-linux-compat-01-skip-compat_h-version-checks.patch]: Temporary fix when building 5.6 for Geneirc:

https://github.com/LibreELEC/LibreELEC.tv/pull/4199/commits/436837112c24eafa9d83614a7374c650d26f0d2c

**Comment by chewitt** [comment]
I tweaked the patch so it only contains an override for the version warning (text below is correct, support was merged in 5.6) and I've clean built this against master HEAD and 5.6-rc2 kernel.

**Comment by MilhouseVH** [comment]
Did you build this against 5.6-rc2 without https://github.com/LibreELEC/LibreELEC.tv/commit/436837112c24eafa9d83614a7374c650d26f0d2c because this PR is/was failing for me (I need to test it again now you've updated the patch). The 5.6-rc2 build appeared to be failing for reasons other than the patch and it looked like need to avoid building the compat module with 5.6-rc2+. I'll test again later this morning.

**Pull request merged**